usergroup_permissions.py

```diff
@@ -0,0 +1,41 @@
+
+from urllib.parse import urljoin
+
+
+class Permissions:
+    def __init__(self, logger, sharepoint_client):
+        self.logger = logger
+        self.sharepoint_client = sharepoint_client
+        self.logger.info('Initilized Permissions class')
+
+    def check_permissions(self, key, rel_url, title=None, id=None):
```

Comment:
too verbose, you can create a mapping like this:

```
maps = {'site': '/_api/web/HasUniqueRoleAssignments', 
  'lists': '/_api/web/lists/getbytitle({0})/HasUniqueRoleAssignments',
...
}
```

and then your function is a one-liner

```
unique = self.sharepoint_client.get(urljoin(rel_url, maps[key], query='?select=title')
```

Reply:
Done